MinecraftClient/ChatBots/MineCube.cs

```diff
@@ -0,0 +1,517 @@
+                        {
+                            tempRow.BlocksToMine.Add(tempLocation);
+                        }
+                    }
+                    if (tempRow.BlocksToMine.Count > 0)
+                    {
+                        tempLayer.addRow(tempRow);
+                    }
+                }
+                if (tempLayer.RowsToMine.Count > 0)
+                {
+                    cubeToMine.addLayer(tempLayer);
+                }
+            }
+
+            // Remove later ;D
+            printCubeToConsole(cubeToMine);
+
+            if (Settings.DebugMessages)
+            {
+                printCubeToConsole(cubeToMine);
+            }
+
+            return cubeToMine;
+        }
+
+        /// <summary>
+        /// Get all numbers between from and to.
+        /// </summary>
+        /// <param name="start">Number to start</param>
+        /// <param name="end">Number to stop</param>
+        /// <returns>All numbers between the first, including the stop number</returns>
+        public List<int> getNumbersFromTo(int start, int stop)
+        {
+            List<int> tempList = new List<int>();
+            if (start <= stop)
+            {
+                for (int i = start; i <= stop; i++)
+                {
+                    tempList.Add(i);
+                }
+            }
+            else
+            {
+                for (int i = start; i >= stop; i--)
+                {
+                    tempList.Add(i);
+                }
+            }
+            return tempList;
+        }
+
+        public Func<Location, Location> getHeadLocation = locFeet => new Location(locFeet.X, locFeet.Y + 1, locFeet.Z);
+
+        /// <summary>
+        /// Checks whether a material is minable
+        /// </summary>
+        /// <param name="block">Block that should be checked</param>
+        /// <returns>Is block minable</returns>
+        private bool isMinable(Material block)
+        {
+            return (
+                block != Material.Air &&
+                block != Material.Bedrock &&
+                !block.IsLiquid()
+                    );
+        }
+
+        private void selectCorrectSlotInHotbar(ItemType[] tools)
+        {
+            if (GetInventoryEnabled())
+            {
+                foreach (ItemType tool in tools)
+                {
+                    int[] tempArray = GetPlayerInventory().SearchItem(tool);
+                    if(tempArray.Length > 0 && tempArray[0] < 10)
+                    {
+                        ChangeSlot(Convert.ToInt16(tempArray[0]));
+                        break;
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Evaluates the right tool for the job
+        /// </summary>
+        /// <param name="block">Enter the Material of a block</param>
+        /// <returns>The right tool cathegory as string</returns>
+        private ItemType[] getCorrectToolForBlock(Material block)
```

Comment:
```suggestion
        private ItemType[] GetCorrectToolForBlock(Material block)
```

***

You may want to move this method to [MaterialExtensions.cs](https://github.com/ORelio/Minecraft-Console-Client/blob/master/MinecraftClient/Mapping/MaterialExtensions.cs)